parsers/IE.py

```diff
@@ -0,0 +1,279 @@
+from datetime import datetime, timedelta
+from logging import Logger, getLogger
+from typing import Optional
+
+import arrow
+import pytz
+from requests import Response, Session
+
+from parsers.lib.config import refetch_frequency
+from parsers.lib.exceptions import ParserException
+
+IE_TZ = pytz.timezone("Europe/Dublin")
+
+DEMAND_URL = "https://www.smartgriddashboard.com/DashboardService.svc/data?area=demandactual&region={zone}&datefrom={dt}+00%3A00&dateto={dt}+23%3A59"
+DEMAND_FORECAST_URL = "https://www.smartgriddashboard.com/DashboardService.svc/data?area=demandforecast&region={zone}&datefrom={dt}+00%3A00&dateto={dt}+23%3A59"
+WIND_URL = "https://www.smartgriddashboard.com/DashboardService.svc/data?area=windactual&region={zone}&datefrom={dt}+00%3A00&dateto={dt}+23%3A59"
+WIND_FORECAST_URL = "https://www.smartgriddashboard.com/DashboardService.svc/data?area=windforecast&region={zone}&datefrom={dt}+00%3A00&dateto={dt}+23%3A59"
+EXCHANGE_URL = "https://www.smartgriddashboard.com/DashboardService.svc/data?area=interconnection&region={zone}&datefrom={dt}+00%3A00&dateto={dt}+23%3A59"
+
+KINDS_URL = {
+    "demand": DEMAND_URL,
+    "demand_forecast": DEMAND_FORECAST_URL,
+    "wind": WIND_URL,
+    "wind_forecast": WIND_FORECAST_URL,
+    "exchange": EXCHANGE_URL,
+}
+
+ZONE_MAPPING = {
+    "IE": {"key": "ROI", "exchange": "INTER_EWIC"},
+    "GB-NIR": {"key": "NI", "exchange": "INTER_MOYLE"},
+}
+
+
+def fetch_data(
+    target_datetime: Optional[datetime],
+    zone_key: str,
+    kind: str,
+    session: Session,
+) -> list:
+    """
+    Gets values and corresponding datetimes for the specified data kind in ROI.
+    Removes any values that are in the future or don't have a datetime associated with them.
+    """
+    assert type(target_datetime) == datetime
+    assert kind != ""
+    assert session is not None
+
+    resp: Response = session.get(
+        KINDS_URL[kind].format(
+            dt=target_datetime.strftime("%d-%b-%Y"), zone=ZONE_MAPPING[zone_key]["key"]
+        )
+    )
+    try:
+        data = resp.json().get("Rows", {})
+    except:
+        raise ParserException(
+            parser="IE.py",
+            message=f"{target_datetime}: {kind} data is not available for {zone_key}",
+        )
+    filtered_data = [item for item in data if item["Value"] is not None]
+    return filtered_data
+
+
+@refetch_frequency(timedelta(days=1))
+def fetch_production(
+    zone_key: str,
+    session: Session = Session(),
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> list:
+    """Gets values for wind production and estimates unknwon production as demand - wind - exchange"""
+    if target_datetime is None:
+        target_datetime = datetime.now().replace(tzinfo=IE_TZ)
+
+    if session is None:
+        session = Session()
```

Comment:
```suggestion
```
We can remove this now since session have a default argument of `Session()` with the previous changes, sorry if I missed it before.